qa/L0_decoupled/test.sh

```diff
@@ -48,78 +48,91 @@ rm -f *.log
 
 CLIENT_LOG=`pwd`/client.log
 DATADIR=/data/inferenceserver/${REPO_VERSION}/qa_model_repository
-MODELDIR=`pwd`/models
 SERVER=/opt/tritonserver/bin/tritonserver
 SERVER_ARGS="--model-repository=`pwd`/models"
 SERVER_LOG="./inference_server.log"
 source ../common/util.sh
 
-cp -r $DATADIR/libtorch_nobatch_int32_int32_int32 $MODELDIR/.
-(cd $MODELDIR/libtorch_nobatch_int32_int32_int32 && \
-    sed -i "s/dims:.*\[.*\]/dims: \[ 1 \]/g" config.pbtxt)
 
-run_server
-if [ "$SERVER_PID" == "0" ]; then
-    echo -e "\n***\n*** Failed to start $SERVER\n***"
-    cat $SERVER_LOG
-    exit 1
-fi
-
-for i in \
-            test_one_to_none \
-            test_one_to_one \
-            test_one_to_many \
-            test_no_streaming \
-            test_response_order \
-            test_wrong_shape ; do
-
-    echo "Test: $i" >>$CLIENT_LOG
-    set +e
-    python $DECOUPLED_TEST DecoupledTest.$i >>$CLIENT_LOG 2>&1
-    if [ $? -ne 0 ]; then
-            echo -e "\n***\n*** Test $i Failed\n***" >>$CLIENT_LOG
-            echo -e "\n***\n*** Test $i Failed\n***"
-            RET=1
-    else
-        check_test_results $TEST_RESULT_FILE 1
-        if [ $? -ne 0 ]; then
-            cat $CLIENT_LOG
-            echo -e "\n***\n*** Test Result Verification Failed\n***"
-            RET=1
-        fi
-    fi
-    set -e
+TRIALS="python custom"
+
+for trial in $TRIALS; do
+  if [ $trial == "python" ]; then
+    MODELDIR=`pwd`/python_models
+  else
+    MODELDIR=`pwd`/models
+  fi
+
+  SERVER_ARGS="--model-repository=$MODELDIR"
+  cp -r $DATADIR/libtorch_nobatch_int32_int32_int32 $MODELDIR/.
+  (cd $MODELDIR/libtorch_nobatch_int32_int32_int32 && \
+   sed -i "s/dims:.*\[.*\]/dims: \[ 1 \]/g" config.pbtxt)
+
+  run_server
+  if [ "$SERVER_PID" == "0" ]; then
+      echo -e "\n***\n*** Failed to start $SERVER\n***"
+      cat $SERVER_LOG
+      exit 1
+  fi
+  
+  for i in \
+              test_one_to_none \
```

Comment:
Doesn't it make sense to add shared memory leak detector check on these tests?

Reply:
I wonder what is the best way to detect memory leak is though.

Reply:
I can give it a try and if it took a lot of time. I'll do it separately.

Reply:
It looks like there is not any shared memory growth issues but the shared memory leak detector fails because of some underlying object that is in scope (perhaps by the response threads) that makes it fail to match the exact value. I logged the free shared memory values as the test ran and it doesn't look like there is any leak:
![image](https://user-images.githubusercontent.com/10105175/165619381-e0c045c6-3cc9-4cf4-bd33-98752579b639.png)

I think we might need to change our leak detection tests to be a bit more forgiving. @GuanLuo do you think it makes sense to add shared memory usage monitoring to the long running tests so that we can detect any unbounded growth?